### PR TITLE
TF-1810 Store email when receive in foreground

### PR DESCRIPTION
**Pull request by dab246**
### Issue

#1810 

### Resolved

https://github.com/linagora/tmail-flutter/assets/80730648/e6d4c7e4-2797-430a-803f-e88e68a590e4

### CODE REVIEW

File: lib/features/caching/utils/caching_constants.dart [hunk 1]
```diff
@@ -20,4 +20,6 @@ class CachingConstants {
 
   static const String fcmCacheBoxName = 'fcm_cache_box';
   static const String detailedEmailCacheBoxName = 'detailed_email_cache_box';
+
+  static const int maxNumberRecentEmails = 10;
```
[the remark this answers is not in the document]
Comment (hoangdat) [reply lib/features/caching/utils/caching_constants.dart:26]: `maxNumberOpenedEmailsForOffline = 30`

File: lib/features/email/data/datasource/email_datasource.dart [hunk 2]
```diff
@@ -72,4 +73,10 @@ abstract class EmailDataSource {
   Future<List<EmailId>> deleteMultipleEmailsPermanently(Session session, AccountId accountId, List<EmailId> emailIds);
 
   Future<bool> deleteEmailPermanently(Session session, AccountId accountId, EmailId emailId);
+
+  Future<void> storeDetailedEmailToCache(Session session, AccountId accountId, DetailedEmail detailedEmail);
```
Comment (hoangdat) [inline lib/features/email/data/datasource/email_datasource.dart:77]:
```suggestion
  Future<void> storeDetailedEmail(Session session, AccountId accountId, DetailedEmail detailedEmail);
```

File: lib/features/email/data/datasource/email_datasource.dart [hunk 3]
```diff
@@ -72,4 +73,10 @@ abstract class EmailDataSource {
...
+
+  Future<Email> getDetailedEmailById(Session session, AccountId accountId, EmailId emailId);
+
+  Future<void> storeEmailToCache(Session session, AccountId accountId, Email email);
```
Comment (hoangdat) [inline lib/features/email/data/datasource/email_datasource.dart:81]:
```suggestion
  Future<void> storeEmail(Session session, AccountId accountId, Email email);
```

**Review by hoangdat — COMMENTED** [review]
Please check web app to make sure this PR don't make any impact

File: lib/features/caching/utils/caching_constants.dart [hunk 4]
```diff
@@ -22,4 +24,7 @@ class CachingConstants {
   static const String detailedEmailCacheBoxName = 'detailed_email_cache_box';
 
   static const int maxNumberRecentEmails = 10;
+
+  static const String appFolderName = AppConstants.applicationName;
```
[the remark this answers is not in the document]
Comment (dab246) [reply lib/features/caching/utils/caching_constants.dart:28]: Ok. I will remove it.

**Review by nqhhdev — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**